packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -74,10 +77,13 @@ export class IamPolicyChecksWebview extends VueWebview {
             const customChecksTextArea = await _readCustomChecksFile(path)
             this.onFileReadError.fire('') // Reset error message display if no error is found
             return customChecksTextArea
-        } catch (err) {
+        } catch (err: any) {
             if (err instanceof PolicyChecksError && (err.code as PolicyChecksErrorCode) === 'FileReadError') {
                 this.onFileReadError.fire(err.message)
             }
+            telemetry.accessanalyzer_iamPolicyChecksError.emit({
```

Comment:
instead of using `.emit()`, use `.record()`. This will handle errors automatically. There are many examples in the codebase and it is also explained in `telemetry.md`.

Applicable throughout PR.

Reply:
One of the reasons I strayed away from `.record()` is because the tool we use returns an "error" when we discover findings, under a different exit code. We check for the exit code to handle it gracefully, but if I used `.run()` and `.record()` there is no way for me to specify that it actually `Succeeded`.
I found that I am unable to set the `result` or `reason` when using `.record()`

Reply:
Using `.record()` now